### PR TITLE
Add per-step timing information

### PR DESCRIPTION
**Pull request by Flamefire**
Shows e.g.:
```
== fetching files...
== creating build dir, resetting environment...
== ... (took 2 secs)
== unpacking...
== ... (took 15 secs)
== patching...
== ... (took 7 secs)
```

Requires Python 2.7/3.2

### CODE REVIEW

**Comment by Flamefire** [comment]
@smoors Maybe you got time for this too?

File: easybuild/tools/utilities.py [hunk 1]
```diff
@@ -300,22 +300,21 @@ def time2str(delta):
...
-        secs = int(delta_secs - (hours * 3600) - (mins * 60))
-        hours_str = 'hours' if hours > 1 else 'hour'
-        res = '%d %s %d min %d sec' % (hours, hours_str, mins, secs)
+    delta_secs = delta.total_seconds()
 
-    return res
+    hours = int(delta_secs / 3600)
+    delta_secs -= hours * 3600
+    mins = int(delta_secs / 60)
+    secs = int(delta_secs - mins * 60)
```
Comment (smoors) [inline easybuild/tools/utilities.py:308]: can be even more concise:
```suggestion
    hours, remainder = divmod(s, 3600)
    mins, secs = divmod(remainder, 60)

```

File: easybuild/tools/utilities.py [hunk 2]
```diff
@@ -300,22 +300,21 @@ def time2str(delta):
...
+
+    res = []
+    if hours:
+        res.append('%d %s' % (hours, 'hour' if hours == 1 else 'hours'))
+    if mins or hours:
+        res.append('%d %s' % (mins, 'min' if mins == 1 else 'mins'))
+    res.append('%d %s' % (secs, 'sec' if secs == 1 else 'secs'))
```
Comment (smoors) [inline easybuild/tools/utilities.py:315]: I suspect this is going to break some users' grep scripts :)

**Review by smoors — COMMENTED** [review]
minor suggestion

File: easybuild/tools/utilities.py (hunk 2 above)
Comment (Flamefire) [reply easybuild/tools/utilities.py:313]: Well, I hope they don't grep for the "secs" part, but for "FINISHED IN" (or what that was) ;)

File: easybuild/tools/utilities.py (hunk 1 above)
Comment (Flamefire) [reply easybuild/tools/utilities.py]: I like it, thanks! And no, the results of divmod are already ints :)

Comment (smoors) [reply easybuild/tools/utilities.py]: oh it depends on the type of delta_secs. if it's float, the result is also float ;)

Comment (Flamefire) [reply easybuild/tools/utilities.py]: Oh, yes. It's a float here... :/
Then you are right.

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @Flamefire!

**Pull request merged**